### ISSUE TITLE
Escape key and fullscreen mode

### ISSUE THREAD

**Issue by Archie-2021** [zsviczian/obsidian-excalidraw-plugin#309]
**Is your feature request related to a problem? Please describe.**
when I cancel commands with Escape key in fullscreen mode it causes to the app jump out of fullscreen. 
and thanks for this awesome plugin
**Describe the solution you'd like**
one of these shortcuts should be changed.

**Describe alternatives you've considered**

**Additional context**

**Comment by zsviczian** [comment]
Yes, full screen mode needs a little work. Other issues with fullscreen mode include command palette not being visible. I haven’t had the time / energy yet to dive deeper into debugging fullscreen mode. 

How would you want to close fullscreen mode instead of ESC?

**Comment by Archie-2021** [comment]
> Yes, full screen mode needs a little work. Other issues with fullscreen mode include command palette not being visible. I haven’t had the time / energy yet to dive deeper into debugging fullscreen mode.
> 
> How would you want to close fullscreen mode instead of ESC?

Tbh I am not sure what is the best option as ESC is the best option for that, maybe changing it to F1 or Ctrl+F1 or something like that. Of course the best option would be to customize it in hotkeys

**Comment by zsviczian** [comment]
Based on the information I found on stack overflow, ESC will always close fullscreen.

_Any browsers complying to the standards will not (and should not) let you block this. Quite simply because it would be very intrusive and will confuse the users, who expect to always have a clear "panic button" to escape a screen._

https://stackoverflow.com/questions/21279174/prevent-esc-from-exiting-fullscreen-mode

I have, however, implemented a few improvements to fullscreen mode:
- command palette is now displayed correctly
- modal windows initiated from Excalidraw Script Engine will show correctly
- fullscreen is even more full screen, with no bezels, no padding, etc.

The improvements will be released with 1.5.7 later today.

**Comment by Archie-2021** [comment]
> Based on the information I found on stack overflow, ESC will always close fullscreen.
> 
> _Any browsers complying to the standards will not (and should not) let you block this. Quite simply because it would be very intrusive and will confuse the users, who expect to always have a clear "panic button" to escape a screen._
> 
> https://stackoverflow.com/questions/21279174/prevent-esc-from-exiting-fullscreen-mode
> 
> I have, however, implemented a few improvements to fullscreen mode:
> 
> * command palette is now displayed correctly
> * modal windows initiated from Excalidraw Script Engine will show correctly
> * fullscreen is even more full screen, with no bezels, no padding, etc.
> 
> The improvements will be released with 1.5.7 later today.

Thats what i guessed too and makes sense. Is it possible then to find another key for canceling? for example, lets say I am drawing a line and I didn't double click the last point because I was not at the moment it is going to be the last one. How can I cancel this without using ESC? 

thanks, specially command pallet being available in fullscreen is a huge help

**Comment by zsviczian** [comment]
CTRL+Enter has the same effect as ESC.

You can also create an ExcalidrawAutomate Script and assign a hotkey. The following script will cancel the editing of the currently edited element and delete what you have edited from the scene.

```javascript
st = ea.getExcalidrawAPI().getAppState();
el = ea.getViewElements().filter((el)=>el.id !== st.editingElement.id);
st.editingElement=null;
st.draggingElement = null;
ea.selectedElementIds={};
ea.getExcalidrawAPI().updateScene({elements:el,appState:st});
```
